.github/workflows/promotion-check.yml

```
@@ -15,43 +16,71 @@ jobs:
     runs-on: self-hosted
     if: github.event.pull_request.head.repo.full_name == github.repository
     steps:
-      - name: Verify PR source is staging
+      - name: Verify PR source branch
         run: |
```

actionlint (1.7.12)

[error] "github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)